### ISSUE TITLE
Feature request: inserting and evaluation of code

### ISSUE THREAD

**Issue by uncomfyhalomacro** [Keats/gutenberg#2050]
Hello. New user of zola here. Is there any possibility to have this feature in zola?
Link to feature: https://franklinjl.org/code/

If it is possible, are the maintainers/devs free to add this feature? If not, maybe I can get some help in adding this feature? I am not eloquent in Rust but I am willing to be guided on where to start. Thanks a bunch for this project. It got me to write my first static blog site.

**Comment by c-git** [comment]
Could you explain a bit more which part of the link you are referring to? Is it the ability to run code and show the output?

**Comment by uncomfyhalomacro** [comment]
> Could you explain a bit more which part of the link you are referring to? Is it the ability to run code and show the output?

Yes it is the part where you can run code and show the output. Like a mini playground.
- https://franklinjl.org/code/#inserting_code
- https://franklinjl.org/code/#evaluating_code
- https://franklinjl.org/code/#inserting_the_output

But instead of Julia, this feature should be language-agnostic. The idea is the static site generator will capture the output from a script/binary during the generation of the website and show it in the generated static website as some fancy code block.

I am not sure if this violates the philosophy of zola as I am not familiar with its development.

**Comment by c-git** [comment]
I see thank you for explaining. I was just trying to understand to see if this was something that could already be done. We'll have to wait for when the maintainer gets a chance to respond to this to see what he thinks.

**Comment by davidmreed** [comment]
It seems plausible to me that you could implement this using the [`load_data()`](https://www.getzola.org/documentation/templates/overview/#load-data) function, since it can call arbitrary web services, _if_ you have or build a web service that can provide this kind of REPL functionality.

Needless to say it's a high-risk activity to run arbitrary user-supplied code, and that capability would massively increase the attack surface of a static site generator.

**Comment by uncomfyhalomacro** [comment]
> It seems plausible to me that you could implement this using the [`load_data()`](https://www.getzola.org/documentation/templates/overview/#load-data) function, since it can call arbitrary web services, _if_ you have or build a web service that can provide this kind of REPL functionality.
> 
> Needless to say it's a high-risk activity to run arbitrary user-supplied code, and that capability would massively increase the attack surface of a static site generator.

Ah that is true. However I do believe that in Franklin.jl's case, I think it only does a fake Live Code Evaluation. Because the way Franklin does it is, it runs the code, gets the output. This output is saved to a file so that if a user clicks a sample code, it does not really run the code, it just shows the output of the code mimicking a live run of the code. As explained in https://franklinjl.org/code/#live_evaluation_julia.

In summary, live code evaluation is _actually done locally_ when building the site, and the output is saved into a file to mimick live code evaluation when opening the site.

**Comment by Keats** [comment]
Right now there's no plan to have Zola call other binaries.
You could do [Offline evaluation (any language)](https://franklinjl.org/code/#offline_evaluation_any_language) by using a shortcode I think?
